ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -885,6 +885,141 @@ public void testExpiryAfterMetadataExchange() throws Exception {
+            break;
+          case "A2P":
+            addPutMessagesToReplicasOfPartition(Collections.singletonList(blobIdV5),
+                Collections.singletonList(remoteHost));
+            break;
+          case "A2D":
+            addDeleteMessagesToReplicasOfPartition(partitionId, blobIdV5, Collections.singletonList(remoteHost));
+            break;
+        }
+      }
+      // Do the replica metadata exchange.
+      List<ReplicaThread.ExchangeMetadataResponse> responses =
+          replicaThread.exchangeMetadata(new MockConnection(remoteHost, 10, remoteConversionMap), singleReplicaList);
+      // Do Get request to fix missing keys
+      replicaThread.fixMissingStoreKeys(new MockConnection(remoteHost, 10, remoteConversionMap), singleReplicaList,
+          responses);
+
+      // Verify
+      String resultStr = caseAndResult.getValue();
+      String[] expectedResults = resultStr.equals("") ? new String[0] : resultStr.split(" ");
+      int size = localHost.infosByPartition.get(partitionId).size();
+      assertEquals("Mismatch in number of messages on local host after replication", expectedResults.length, size);
+      for (int i = 0; i < size; ++i) {
+        switch (expectedResults[i]) {
+          case "A1P":
+            assertEquals("Mismatch in blodId on local host after replication", blobIdV2.toString(),
+                localHost.infosByPartition.get(partitionId).get(i).getStoreKey().toString());
+            assertFalse("Mismatch in message type on local host after replication",
+                localHost.infosByPartition.get(partitionId).get(i).isDeleted());
+            break;
+          case "A1D":
+            assertEquals("Mismatch in blodId on local host after replication", blobIdV2.toString(),
+                localHost.infosByPartition.get(partitionId).get(i).getStoreKey().toString());
+            assertTrue("Mismatch in message type on local host after replication",
+                localHost.infosByPartition.get(partitionId).get(i).isDeleted());
+            break;
+          case "A2P":
+            assertEquals("Mismatch in blodId on local host after replication", blobIdV5.toString(),
+                localHost.infosByPartition.get(partitionId).get(i).getStoreKey().toString());
+            assertFalse("Mismatch in message type on local host after replication",
+                localHost.infosByPartition.get(partitionId).get(i).isDeleted());
+            break;
+          case "A2D":
+            assertEquals("Mismatch in blodId on local host after replication", blobIdV5.toString(),
+                localHost.infosByPartition.get(partitionId).get(i).getStoreKey().toString());
+            assertTrue("Mismatch in message type on local host after replication",
+                localHost.infosByPartition.get(partitionId).get(i).isDeleted());
+            break;
+        }
+      }
+
+      // Clean up partitions on both local and remote
+      localHost.clearAll();
```

Comment:
Instead of having to clean up after every test case, might be cleaner to structure this as multiple tests with a `setup(...); testCase = ...; testAndVerify(testCase);` sort of set up, with each test having the same three line setup but with a different testCase.  This is also preferable as we can see if multiple test cases break (as opposed to failing out if one case fails) and it's easier to see which test case failed.

Generally having lots of small tests is preferable to having a few big tests that each test a lot of things.